pylabrobot/config/service/writer.py

```diff
@@ -0,0 +1,18 @@
+from abc import ABC, abstractmethod
+from typing import TypeVar, Generic
+
+from pylabrobot.config.config import Config
+
+T = TypeVar('T')
+
+
+class ConfigWriter(Generic[T], ABC):
+  """ConfigWriter is an abstract class for writing a Config object to a
+  stream."""
+
+  open_mode: str = "w"
+
+  @abstractmethod
+  def write(self, w: T, cfg: Config):
+    """Write a Config object."""
+    raise NotImplementedError
```

Comment:
is raise needed when method is marked as abstarctmethod?

Reply:
Probably not, I got into the habit from a book I read which uses both to be explicit https://github.com/cosmicpython/code/blob/master/src/allocation/adapters/repository.py

Happy to take the raise out if you prefer!

Reply:
is explicitness the only motivation? In that case I think just "saying it once" is clearer. Happy to discuss